mlflow/tracing/client.py

```diff
@@ -362,7 +364,17 @@ def download_trace_extra_fields(
         next_max_results = max_results
         next_token = page_token
 
-        executor = ThreadPoolExecutor() if include_spans else nullcontext()
+        # Get configured max workers from environment variable or use default
+        # Use minimum of 32 threads or 4x CPU count (whichever is larger)
+        # This is important because these threads are network IO-bound (waiting for downloads)
+        # rather than CPU-bound, so we want more threads than CPU cores to maximize throughput
+        configured_max_workers = MLFLOW_SEARCH_TRACES_MAX_THREADS.get()
+        max_workers = (
+            configured_max_workers
+            if configured_max_workers is not None
+            else max(32, os.cpu_count() * 4)
```

Comment:
```suggestion
            else max(32, (os.cpu_count() or 1) * 4)
```

`cpu_count` can return `None`.

Reply:
Oh wow, thanks Haru!